libs/designer/src/lib/core/state/connection/connectionSelector.ts

```diff
@@ -46,6 +46,8 @@ export const useGateways = (subscriptionId: string, connectorName: string) => {
 
 export const useSubscriptions = () => useQuery('subscriptions', async () => GatewayService().getSubscriptions());
 
+export const useGatewayServiceConfig = () => useQuery('gatewayServiceConfig', async () => GatewayService().getConfig?.() ?? {});
```

Comment:
This doesn't need to be a `useQuery`, we can just make it a `useMemo` since we're not actually querying data, it's just a read of the config from the service

```suggestion
export const useGatewayServiceConfig = () => useMemo(() => GatewayService().getConfig?.() ?? {}, []);
```